sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientProvider.java

```diff
@@ -8,21 +8,27 @@
 import com.azure.core.util.HttpClientOptions;
 import reactor.netty.resources.ConnectionProvider;
 
+import java.util.concurrent.atomic.AtomicReference;
+
 /**
  * An {@link HttpClientProvider} that provides an implementation of HttpClient based on Netty.
  */
 public final class NettyAsyncHttpClientProvider implements HttpClientProvider {
+    private static final AtomicReference<HttpClient> DEFAULT_HTTP_CLIENT = new AtomicReference<>();
 
     @Override
     public HttpClient createInstance() {
-        return new NettyAsyncHttpClientBuilder().build();
+        // by default use a singleton instance of http client
+        DEFAULT_HTTP_CLIENT.compareAndSet(null, new NettyAsyncHttpClientBuilder().build());
```

Comment:
Since the defaults for max connections, queue size, etc are now going to be shared across multiple Azure data-plane clients, it could potentially break some existing applications that created multiple clients and used up resources close to the default.

Reply:
On the other hand sharing single httpclient/connection pool is challenging given current default. Which is a trap for new users who don't (or can't) setup application wide scope for the clients.
How about introducing compat switch (sys property) to provide an easy way to fallback to previous default?